### PR TITLE
lighter rejection of type variable instantiation

### PR DESCRIPTION
**Pull request by nikivazou**
Implementation of Ranjit's suggestion in #1553. Let's see what breaks now.

### CODE REVIEW

**Comment by nikivazou** [comment]
@ranjitjhala circleci and I are quite confident on this. Let me know if you want to wait for your review again. 

I am not sure if this edit only applies on assumed functions (as is now) or should also apply for the imported, but we can discuss this on a next call.

File: src/Language/Haskell/Liquid/Types/Types.hs
Comment (ranjitjhala) [inline src/Language/Haskell/Liquid/Types/Types.hs]: Can you add some documentation here to explain what this field means?eg an example of when the field is true/false?

**Review by ranjitjhala — CHANGES_REQUESTED** [review]
This looks great (assuming tests pass etc.) looks like very few changes? (Ie additions of assume?)

But can you add the comment about the field before merging? Thanks!

**Comment by ranjitjhala** [comment]
Hi Niki, this looks good right it doesn’t break all the tests and so on? I
left one comment that is could you please add some description of what
happens at that field? Just for documentation

File: src/Language/Haskell/Liquid/Types/Types.hs
Comment (nikivazou) [reply src/Language/Haskell/Liquid/Types/Types.hs]: added documentation!

**Review by ranjitjhala — APPROVED** [review]
Great, thanks!

File: benchmarks/popl18/ple/pos/BasicLambdas.hs
Comment (ranjitjhala) [inline benchmarks/popl18/ple/pos/BasicLambdas.hs:51]: Do you know why this test fails? (No need to hold up the merge, just curious?)

**Comment by ranjitjhala** [comment]
@nikivazou -- looks good to me, thanks! this looks like it worked out quite well? (A little surprisingly well actually, but good you added the neg tests...)

File: benchmarks/popl18/ple/pos/BasicLambdas.hs
Comment (nikivazou) [reply benchmarks/popl18/ple/pos/BasicLambdas.hs:51]: I have removed support for alpha equivalence, there is not reason why it passed. 
Before it was ansoundly passing because extensionality was not checking its proof argument at all.

**Comment by nikivazou** [comment]
> @nikivazou -- looks good to me, thanks! this looks like it worked out quite well? (A little surprisingly well actually, but good you added the neg tests...)

Well, it basically addresses exactly the source of inconsistency of these two tests. 
Now, it the inconsistency can be generated differently, I am not sure...

**Pull request merged**
